crates/dojo-core/src/world.cairo

```diff
@@ -444,6 +451,16 @@ mod world {
             let upgradeable_dispatcher = IUpgradeableDispatcher { contract_address };
             upgradeable_dispatcher.upgrade(class_hash);
 
+            if self.initialized_contract.read(contract_address.into()) {
+                panic!("Already initialized");
```

Comment:
We may try to have more detailed error message, as now it's a string into the panic we may even format to have the name of the contract. WDYT?